diff --git a/src/Makefile b/src/Makefile
index ace912d1..364bc3b7 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -248,10 +248,11 @@ ifeq ($(UNAME),Linux)
 CFLAGS_NATIVE           := $(CFLAGS)
 CFLAGS_NATIVE           += -DWITH_HWMON
 LFLAGS_NATIVE           := $(LFLAGS)
-LFLAGS_NATIVE           += -lpthread
+#LFLAGS_NATIVE           += -lpthread
 LFLAGS_NATIVE           += -ldl
-LFLAGS_NATIVE           += -lrt
+#LFLAGS_NATIVE           += -lrt
 LFLAGS_NATIVE           += -lm
+LFLAGS_NATIVE           += /data/data/com.termux/files/usr/lib/libiconv.a
 endif # Linux
 
 ifeq ($(UNAME),FreeBSD)
@@ -578,7 +579,7 @@ CFLAGS_CROSS_WIN        += -I$(WIN_ICONV)/include/
 CFLAGS_CROSS_WIN        += -DWITH_HWMON
 
 LFLAGS_CROSS_LINUX      := $(LFLAGS)
-LFLAGS_CROSS_LINUX      += -lpthread
+#LFLAGS_CROSS_LINUX      += -lpthread
 LFLAGS_CROSS_LINUX      += -ldl
 LFLAGS_CROSS_LINUX      += -lm
 LFLAGS_CROSS_WIN        := $(LFLAGS)
diff --git a/src/affinity.c b/src/affinity.c
index 0cf6d843..4c5f968a 100644
--- a/src/affinity.c
+++ b/src/affinity.c
@@ -47,7 +47,7 @@ typedef cpuset_t cpu_set_t;
 
 int set_cpu_affinity (MAYBE_UNUSED hashcat_ctx_t *hashcat_ctx)
 {
-#if defined (__CYGWIN__)
+#if defined (__CYGWIN__) || defined(__ANDROID__)
   return 0;
 #else
   const user_options_t *user_options = hashcat_ctx->user_options;
